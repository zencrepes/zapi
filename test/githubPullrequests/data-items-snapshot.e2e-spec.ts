import { INestApplication } from '@nestjs/common';
import { GraphQLModule } from '@nestjs/graphql';
import { Test, TestingModule } from '@nestjs/testing';
import { ApolloServerTestClient, createTestClient } from 'apollo-server-testing';
import gql from 'graphql-tag';
import { AppModule } from '../../src/app.module';

//https://github.com/nestjs/nest/issues/484

describe('githubPullrequests/data/items', () => {
  let app: INestApplication;
  let apolloClient: ApolloServerTestClient;

  const GET_ITEMS = gql`
    query($from: Int, $size: Int, $query: String, $sortField: String, $sortDirection: OrderDirection!) {
      githubPullrequests {
        data(query: $query) {
          items(from: $from, size: $size, orderBy: { direction: $sortDirection, field: $sortField }) {
            totalCount
            nodes {
              id
              createdAt
              closedAt
              assignees {
                edges {
                  node {
                    login
                    name
                    avatarUrl
                  }
                }
                totalCount
              }
              reviewRequests {
                totalCount
                edges {
                  node {
                    requestedReviewer {
                      login
                      avatarUrl
                      url
                    }
                  }
                }
              }
              reviews {
                totalCount
                edges {
                  node {
                    author {
                      login
                      url
                      avatarUrl
                    }
                  }
                }
              }
              author {
                url
                avatarUrl
                login
              }
              labels {
                totalCount
                edges {
                  node {
                    name
                    color
                  }
                }
              }
              title
              repository {
                name
                url
                owner {
                  id
                  login
                  url
                }
              }
              milestone {
                title
              }
              state
              url
            }
          }
        }
      }
    }
  `;

  beforeEach(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleFixture.createNestApplication();
    await app.init();

    const module: GraphQLModule = moduleFixture.get<GraphQLModule>(GraphQLModule);
    // apolloServer is protected, we need to cast module to any to get it
    apolloClient = createTestClient((module as any).apolloServer);
  });

  //Assumption is that the first ever PR will never change its content
  it('Get COMMENTS metrics => NO QUERY', async () => {
    const { query } = apolloClient;
    const result: any = await query({
      query: GET_ITEMS,
      variables: {
        from: 0,
        size: 1,
        query: '{}',
        sortField: 'createdAt',
        sortDirection: 'asc',
      },
    });

    expect(result.data.githubPullrequests.data.items.totalCount).toBeGreaterThan(5);

    expect(result.data.githubPullrequests.data.items.nodes[0]).toMatchSnapshot();
  });
});
