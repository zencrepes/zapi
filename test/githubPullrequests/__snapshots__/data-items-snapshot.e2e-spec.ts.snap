// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`githubPullrequests/data/items Get COMMENTS metrics => NO QUERY 1`] = `
Object {
  "assignees": Object {
    "edges": Array [],
    "totalCount": 0,
  },
  "author": Object {
    "avatarUrl": "https://avatars1.githubusercontent.com/u/5667028?u=466c80cbfb8312090daa3bc70c84bd75597532ba&v=4",
    "login": "Fgerthoffert",
    "url": "https://github.com/Fgerthoffert",
  },
  "closedAt": "2018-09-09T23:54:35Z",
  "createdAt": "2018-09-09T23:54:27Z",
  "id": "MDExOlB1bGxSZXF1ZXN0MjE0MTcyMjc1",
  "labels": Object {
    "edges": Array [],
    "totalCount": 0,
  },
  "milestone": null,
  "repository": Object {
    "name": "zencrepes",
    "owner": Object {
      "id": "MDEyOk9yZ2FuaXphdGlvbjQ3MzU2NzU1",
      "login": "zencrepes",
      "url": "https://github.com/zencrepes",
    },
    "url": "https://github.com/zencrepes/zencrepes",
  },
  "reviewRequests": Object {
    "edges": Array [],
    "totalCount": 0,
  },
  "reviews": Object {
    "edges": Array [],
    "totalCount": 0,
  },
  "state": "MERGED",
  "title": "Develop",
  "url": "https://github.com/zencrepes/zencrepes/pull/1",
}
`;
